### PR TITLE
Move all repo loading, configuring into deps.bzl, repos.bzl

### PR DESCRIPTION
**Pull request by fejta**
ref #154

Should make it easier for people to consume this repo.

### CODE REVIEW

**Comment by fejta** [comment]
PASSED ON MY MACHINE (TM)

File: repos.bzl [hunk 1]
```diff
@@ -1234,3 +1238,17 @@ def go_repositories():
         sum = "h1:SvFZT6jyqRaOeXpc5h/JSfZenJ2O330aBsf7JfSUXmQ=",
         version = "v0.0.0-20190308202827-9d24e82272b4",
     )
+
+def configure(minimum_bazel_version = None, rbe_name = "rbe_default", go_version = None, nogo = None, go_modules = go_repositories):
+    if minimum_bazel_version:  # Allow an additional downstrain constraint
```
Comment (BenTheElder) [inline repos.bzl]:
```suggestion
    if minimum_bazel_version:  # Allow an additional downstream constraint
```

Comment (fejta) [reply repos.bzl]: Done

**Comment by fejta** [comment]
HA!!!!!!!!! I HAVE DEFEAT YOU YET AGAIN BAZEL

File: load.bzl [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+def repositories():
+    if not native.existing_rule("bazel_skylib"):
```
Comment (BenTheElder) [inline load.bzl]: For my own understanding: does this pattern mean if `WORKSPACE` has already loaded these rules we won't load a different version?

https://docs.bazel.build/versions/master/skylark/lib/native.html#existing_rule
> Note: If possible, avoid using this function. It makes BUILD files brittle and order-dependent.

😬 

and yet this pattern seems common? https://chromium.googlesource.com/external/github.com/google/protobuf/+/HEAD/protobuf_deps.bzl

File: repos.bzl [hunk 3]
```diff
@@ -1234,3 +1242,18 @@ def go_repositories():
...
+    rbe_autoconfig(name = "rbe_default")
+    protobuf_deps()  # No options
+
+    go_rules_dependencies()  # No options
+    go_register_toolchains(go_version = None, nogo = None)
+
+    gazelle_dependencies()  # TODO(fejta): go_sdk and go_repository_default_cache
+
+    if go_modules:
+        go_modules()
```
Comment (munnerz) [inline repos.bzl]: I don't see `go_modules` set anywhere in this PR - what/how is this intended to be used?

Comment (munnerz) [reply repos.bzl]: Wait, I've been really stupid and not just looked at the default value for this var 🙃

File: load.bzl (hunk 2 above)
Comment (fejta) [reply load.bzl]: Yeah I am just trying to follow standard practice?

https://github.com/bazelbuild/rules_go/blob/master/go/private/repositories.bzl#L44-L47

https://github.com/bazelbuild/bazel-gazelle/blob/master/deps.bzl#L89-L95

Comment (BenTheElder) [reply load.bzl]: ACK just not aware of / certain of the practice :sweat_smile:

**Comment by fejta** [comment]
Anyone want to stamp this? I don't think I have anything we want me to change yet?

**Pull request merged**